homeassistant/components/epic_games_store/calendar.py

```diff
@@ -0,0 +1,108 @@
+"""Calendar platform for a Epic Games Store."""
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import EGSUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the local calendar platform."""
+    coordinator: EGSUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities = [
+        EGSFreeGamesCalendar(coordinator, entry.entry_id),
+        EGSDiscountGameCalendar(coordinator, entry.entry_id),
+    ]
+    async_add_entities(entities, True)
+
+
+class EGSCalendar(CalendarEntity):
+    """A calendar entity by Epic Games Store."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: EGSUpdateCoordinator,
+        unique_id: str,
+        cal_type: str,
+    ) -> None:
+        """Initialize EGSCalendar."""
+        self._cal_type = cal_type
+        self._coordinator = coordinator
+        self._event: CalendarEvent | None = None
+        self._attr_name = f"Epic Games Store {cal_type.title()} Games"
```

Comment:
You can create a device for Epic Games store, and then make the "Free games" and "discount games" translatable